applications/FluidDynamicsApplication/tests/compressible_navier_stokes_symbolic_generator_formulation_test.py

```diff
@@ -0,0 +1,108 @@
+import os
+
+import KratosMultiphysics
+import KratosMultiphysics.KratosUnittest as KratosUnitTest
+
+from KratosMultiphysics.FluidDynamicsApplication.symbolic_generation.compressible_navier_stokes \
+    .compressible_navier_stokes_symbolic_generator import CompressibleNavierStokesSymbolicGenerator
+
+import KratosMultiphysics.FluidDynamicsApplication
+
+class CompressibleNavierStokesSymbolicGeneratorFormulationTest(KratosUnitTest.TestCase):
+    def setUp(self):
+        self.files_to_remove = []
+
+    def tearDown(self):
+        for file_name in self.files_to_remove:
+            KratosMultiphysics.kratos_utilities.DeleteFileIfExisting(file_name)
+        self.files_to_remove = []
+
+    @classmethod
+    def _GetGeneratorSettings(cls, geometry):
+        """Returns the Kratos Parameters for the symbolic generator"""
+        geometry_name = {
+            "2D3N": "triangle",
+            "2D4N": "quadrilateral",
+            "3D4N": "tetrahedron"
+        }[geometry]
+
+        parameters = KratosMultiphysics.Parameters("""
+        {
+            "geometry": "PLEASE SPECIFY A GEOMETRY",
+            "template_filename" : "PLEASE SPECIFY A TEMPLATE FILE",
+            "output_filename"   : "PLEASE SPECIFY AN OUTPUT FILE",
+            "language" : "python"
+        }""")
+
+        parameters["geometry"].SetString(geometry_name)
+        parameters["template_filename"].SetString("test_{}.py_template".format(geometry))
+        parameters["output_filename"].SetString("symbolic_test_{}.py".format(geometry))
+
+        return parameters
+
+    @classmethod
+    def _Generate(cls, geometry):
+        """Generates the code to be tested, and stores it in a file"""
+        params = cls._GetGeneratorSettings(geometry)
+        generator = CompressibleNavierStokesSymbolicGenerator(params)
+        generator.Generate()
+        generator.Write()
+
+        return generator.output_filename
+
+    @classmethod
+    def _ImportSubTestSuite(cls, generated_file_name):
+        """Imports the generated code as a sub-testsuite"""
+        module_name = "compressible_symbolic_generation" + "." + generated_file_name[:-3]
+
+        try:
+            test_module = __import__(module_name, fromlist=module_name)
+        except ModuleNotFoundError as err:
+            raise RuntimeError("Failed to import generated file:", generated_file_name) from err
+        return test_module.SubTestSuite()
+
+
+    def _RunSubTestSuite(self, sub_testsuite, print_results):
+        """Runs the generated code and compares it to the reference"""
+        tests = [subtest_name for subtest_name in dir(sub_testsuite)
+            if callable(getattr(sub_testsuite, subtest_name)) and subtest_name.startswith("test_")]
+
+        for subtest_name in tests:
+            try:
+                result, reference = getattr(sub_testsuite, subtest_name)()
+            except Exception as e:
+                raise RuntimeError("Error in sub-test " + subtest_name) from e
```

Comment:
```suggestion
            self.subTest(subtest_name):
                result, reference = getattr(sub_testsuite, subtest_name)()
```

python already has this :)

=> then you would also not need the custom error message

check it out, I think this works for you

more examples: https://github.com/KratosMultiphysics/Kratos/blob/master/applications/TrilinosApplication/tests/test_trilinos_linear_solvers.py

other than that I don't have more suggestions

Reply:
Oh this is nice. Exactly what I wanted. Thanks.